server/src/main/java/org/opensearch/common/settings/DoubleParser.java

```diff
@@ -0,0 +1,99 @@
+package org.opensearch.common.settings;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.function.Function;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.settings.Setting.Property;
+import org.opensearch.common.settings.Setting.Validator;
+
+public class DoubleParser<T> implements Function<String, T>, Writeable {
+    private Setting<Double> setting;
+    private Double minValue;
+    private Double maxValue;
+    private Double fallback;
+
+    public DoubleParser(Setting<Double> setting) {
+        this.setting = setting;
+    }
+
+    public DoubleParser(StreamInput in) throws IOException {
+        super();
+        // Read the key
+        String key = in.readString();
+        // Read the default value
+        Double defaultValue = in.readDouble();
+        // Read the min value
+        this.minValue = in.readDouble();
+        // Read the max value
+        this.maxValue = in.readDouble();
+        // Read the fallback
+        boolean hasFallback = in.readBoolean();
```

Comment:
I think trying to implement this outside the `Setting` class is overcomplicating things.  Don't worry about the fallbacks for now.  Just focus on replacing the `Function<String, T> parser` argument (`(s) -> parseDouble(s, minValue, maxValue, key, isFiltered(properties))`) with an object.